src/Package.vala

```diff
@@ -283,6 +283,14 @@ public class Eddy.Package : Object {
         int length = int.max (aparts.length, bparts.length);
         for (int i = 0; i < length; i++) {
             int rc = strcmp (aparts[i], bparts[i]);
+            if(i == length -1 ){              
```

Comment:
```suggestion
            if (i == length - 1) {              
```

Reply:
Current version checking uses strcmp only, when it hits a version like .9 comparing it to .10 strcmp will stop at comparing "1" to "9" and return an incorrect up-gradable/down-gradable response. Adding checking of length of the array after last "." we can get around that issue.